test/integration/networks/networks_test.go

```diff
@@ -0,0 +1,255 @@
+			"region1_router1":       fmt.Sprintf("cr-%s-shared-restricted%s-us-west1-cr5", envCode, networkMode),
+			"region1_router2":       fmt.Sprintf("cr-%s-shared-restricted%s-us-west1-cr6", envCode, networkMode),
+			"region2_router1":       fmt.Sprintf("cr-%s-shared-restricted%s-us-central1-cr7", envCode, networkMode),
+			"region2_router2":       fmt.Sprintf("cr-%s-shared-restricted%s-us-central1-cr8", envCode, networkMode),
+			"fw_deny_all_egress":    fmt.Sprintf("fw-%s-shared-restricted-65535-e-d-all-all-all", envCode),
+			"fw_allow_api_egress":   fmt.Sprintf("fw-%s-shared-restricted-65534-e-a-allow-google-apis-all-tcp-443", envCode),
+		},
+	}
+}
+
+func TestNetworks(t *testing.T) {
+
+	orgID := utils.ValFromEnv(t, "TF_VAR_org_id")
+	policyID := getPolicyID(t, orgID)
+	networkMode := getNetworkMode(t)
+
+	restrictedServices := []string{
+		"bigquery.googleapis.com",
+		"storage.googleapis.com",
+	}
+
+	cidrRanges := map[string]map[string][]string{
+		"development": {
+			"base":       []string{"10.0.64.0/21", "10.1.64.0/21"},
+			"restricted": []string{"10.8.64.0/21", "10.9.64.0/21"},
+		},
+		"non-production": {
+			"base":       []string{"10.0.128.0/21", "10.1.128.0/21"},
+			"restricted": []string{"10.8.128.0/21", "10.9.128.0/21"},
+		},
+		"production": {
+			"base":       []string{"10.0.192.0/21", "10.1.192.0/21"},
+			"restricted": []string{"10.8.192.0/21", "10.9.192.0/21"},
+		},
+	}
+
+	googleapisCIDR := map[string]string{
+		"base":       "199.36.153.8/30",
+		"restricted": "199.36.153.4/30",
+	}
+
+	for _, envName := range []string{
+		"development",
+		"non-production",
+		"production",
+	} {
+		t.Run(envName, func(t *testing.T) {
+
+			vars := map[string]interface{}{
+				"access_context_manager_policy_id": policyID,
+			}
+
+			envCode := string(envName[0:1])
+			networks := tft.NewTFBlueprintTest(t,
+				tft.WithTFDir(fmt.Sprintf("../../../3-networks/envs/%s", envName)),
+				tft.WithVars(vars),
+			)
+			networks.DefineVerify(
+				func(assert *assert.Assertions) {
+					// perform default verification ensuring Terraform reports no additional changes on an applied blueprint
+					networks.DefaultVerify(assert)
+
+					servicePerimeterLink := fmt.Sprintf("accessPolicies/%s/servicePerimeters/%s", policyID, networks.GetStringOutput("restricted_service_perimeter_name"))
+					accessLevel := fmt.Sprintf("accessPolicies/%s/accessLevels/%s", policyID, networks.GetStringOutput("restricted_access_level_name"))
+					networkNames := getNetworkResourceNames(envCode, networkMode)
+
+					servicePerimeter := gcloud.Runf(t, "access-context-manager perimeters describe %s --policy %s", servicePerimeterLink, policyID)
+					assert.Equal(servicePerimeterLink, servicePerimeter.Get("name").String(), fmt.Sprintf("service perimeter %s should exist", servicePerimeterLink))
+					var listLevels []string
+					for _, level := range servicePerimeter.Get("status.accessLevels").Array() {
+						listLevels = append(listLevels, level.String())
+					}
```

Comment:
nit: We can use `utils.GetResultStrSlice` here and throughout to get gjson result as a string slice.

https://pkg.go.dev/github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test@v0.0.0-20220505032038-d0a72574cc0b/pkg/utils#GetResultStrSlice

Reply:
thanks for the tip regarding the helper